src/Event/Kafka/KafkaRecord.php

```diff
@@ -0,0 +1,64 @@
+<?php declare(strict_types = 1);
+
+namespace Bref\Event\Kafka;
+
+
+use InvalidArgumentException;
+
+final class KafkaRecord
+{
+
+    private $record;
+
+    /**
+     * @param mixed $record
+     */
+    public function __construct($record)
+    {
+        if (! is_array($record)) {
+            throw new InvalidArgumentException;
+        }
+        $this->record = $record;
+    }
+
+    public function getTopic(): string
+    {
+        return $this->record['topic'];
+    }
+
+    public function getPartition(): int
+    {
+        return $this->record['partition'];
+    }
+
+    public function getOffset(): int
+    {
+        return $this->record['offset'];
+    }
+
+    public function getTimestamp(): int
+    {
+        return $this->record['timestamp'];
+    }
+
+    public function getValue(): mixed
+    {
+        return base64_decode($this->record['value']);
```

Comment:
Is the value always encoded with base64?

Reply:
I haven't found any documentation stating something else. Also unlike api-gateway events, there isn't a property indicating that the payload is base64 encoded or not. I assume the encoding based on the (one and only) example in the documentation: https://docs.aws.amazon.com/lambda/latest/dg/with-kafka.html

Reply:
Oh the docs actually say:

> Each array item contains details of the Kafka topic and Kafka partition identifier, together with a timestamp and a base64-encoded message.

So we're good here, thanks for the link!